pysb/simulator/base.py

```diff
@@ -735,15 +783,230 @@ def observables(self):
+        squeeze (bool), simulations_per_param_set (int), pysb_version (str),
+        creation_date (ISO 8601 format).
+
+        Parameters
+        ----------
+        filename: str
+            Filename to which the data will be saved
+        dataset_name: str or None
+            Dataset name. If None, will default to the name of the simulator
+            class. If the dataset_name already exists within the group,
+            a ValueError is raised.
+        group_name: str or None
+            Group name. If None, will default to the name of the model.
+        append: bool
+            If False, raise IOError if the specified file already exists. If
+            True, append to existing file (or create if it doesn't exist).
+        include_obs_exprs: bool
+            Whether to save observables and expressions in the file or not.
+            If they are not included, they can be recreated from the model
+            and species trajectories when loaded back into PySB, but you may
+            wish to include them for use with external software, or if you
+            have complex expressions which take a long time to compute.
+        """
+        if h5py is None:
+            raise Exception('Please install the h5py package for this feature')
+
+        if self._y is None and not include_obs_exprs:
+            warn('This SimulationResult has no trajectories - '
+                 'you will need to set include_obs_exprs=True if '
+                 'you wish to save observables and expressions')
+
+        if group_name is None:
+            group_name = self._model.name
+        if dataset_name is None:
+            dataset_name = self.simulator_class.__name__
+
+        # As per h5py docs, we should encode strings using numpy.string_
+        # http://docs.h5py.org/en/latest/strings.html
+        enpickle = lambda obj: np.string_(pickle.dumps(obj))
+
+        with h5py.File(filename, 'a' if append else 'w-') as hdf:
+            # Get or create the group
+            try:
+                grp = hdf.create_group(group_name)
+                grp.create_dataset('_model', data=enpickle(self._model))
+            except ValueError:
+                grp = hdf[group_name]
+                model = pickle.loads(grp['_model'][()])
+                if model.name != self._model.name:
+                    raise ValueError('SimulationResult model has name "{}", '
+                                     'but the model in HDF5 file group "{}" '
+                                     'has name "{}"'.format(self._model.name,
+                                                            group_name,
+                                                            model.name))
+
+            # Create the result dataset, which is actually a nested HDF group
+            dset = grp.create_group(dataset_name)
+            if self._y is not None:
+                dset.create_dataset('trajectories', data=self._y,
```

Comment:
I suggest adding `shuffle=True` to all of the `create_dataset` calls, but most of all this one.

Reply:
As a point of reference I ran 5000 simulations of earm_1_0 with slightly perturbed initial conditions, resulting in a `trajectories` dataset of 1000 x 361 x 58, which is 838MB raw. Adding shuffle in addition to gzip saved 9% on the final file size (726MB -> 665MB).